infrastructure/sam/MoralPainFunctions/src/main/java/edu/uva/cs/moralpain/typedb/TypeDBPost.java

```diff
@@ -0,0 +1,86 @@
+import com.vaticle.typedb.client.api.answer.ConceptMap;
+import com.vaticle.typeql.lang.TypeQL;
+import com.vaticle.typeql.lang.query.TypeQLInsert;
+import static com.vaticle.typeql.lang.TypeQL.var;
+
+public class TypeDBPost implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    JSONObject obj = new JSONObject();
+    List<JSONObject> reports = new ArrayList<JSONObject>();
+    JSONParser parser = new JSONParser();
+
+    public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        headers.put("X-Custom-Header", "application/json");
+
+        APIGatewayProxyResponseEvent response = new APIGatewayProxyResponseEvent().withHeaders(headers);
+
+        String body = input.getBody();
+
+        try {
+            JSONObject jsonBodyObject = (JSONObject) parser.parse(body);
+            String id = (String) jsonBodyObject.get("id");
+            @SuppressWarnings("unchecked")
+            List<String> selections = (List<String>) jsonBodyObject.get("selections");
+            Long score_long = (Long) jsonBodyObject.get("score");
+            int score = score_long.intValue();
+            // date comes in as UNIX time in seconds, so we can use int. If it came in
+            // miliseconds, use long.
+            Long timestamp = (Long) jsonBodyObject.get("timestamp");
+
+            String ip = String.format("%s:1729", System.getenv("EC2_IP_ADDRESS"));
+            TypeDBClient client = TypeDB.coreClient(ip);
+            // open up a session
+            try (TypeDBSession session = client.session(System.getenv("DATABASE_NAME"), TypeDBSession.Type.DATA)) {
+
+                try (TypeDBTransaction writeTransaction = session.transaction(TypeDBTransaction.Type.WRITE)) {
+
+                    // Insert a report using a WRITE transaction
+                    var report = var("r").isa("report").has("id", id).has("score", score).has("timestamp",
+                            timestamp);
+
+                    for (String selection : selections) {
+                        report = report.has("selection", selection);
+                    }
+                    TypeQLInsert insertQuery = TypeQL.insert(report);
+
+                    List<Object> insertedId = writeTransaction.query().insert(insertQuery).collect(Collectors.toList());
+                    System.out.println(
+                            "Inserted a person with ID: "
+                                    + ((ConceptMap) insertedId.get(0)).get("r").asThing().getIID());
+
+                    // to persist changes, a write transaction must always be committed (closed)
+                    writeTransaction.commit();
+                }
+            }
+            body = "";
+            return response.withStatusCode(200).withBody(" OK");
+        } catch (ParseException e) {
```

Comment:
probably want to catch more than just parse exceptions

Reply:
What's the best approach when it comes to catching exceptions? As in, how do I know what exceptions might possibly arise and need to be caught, what lines will produce these exceptions, where to look, etc. I'm just wondering about this so I can get better at writing exception handling, since I'm not good with it right now. As I mentioned in a comment before, I tried thinking of ways that might cause errors in the code (like wrong IP address for client, wrong database name, etc), and then ran the code to see what specific errors would be generated (IllegalArgumentException, StatusRuntimeException). Is that kinda how things work?

Reply:
Generally, you don't want to catch the most particular exceptions. When you catch base exceptions you risk swallowing errors you were not expecting.

One way to find the exceptions is trace the source code. Good docs should also tell you which are thrown. Unit testing should exhaust the exceptions as well.

It's also a good idea to log exceptions (at least when the are unexpected).